connectors/migrations/20241212_backfill_github_data_source_folders.ts

```diff
@@ -0,0 +1,236 @@
+import { makeScript } from "scripts/helpers";
+
+import { getGithubCodeDirectoryParentIds } from "@connectors/connectors/github/lib/hierarchy";
+import {
+  getCodeRootInternalId,
+  getDiscussionsInternalId,
+  getIssuesInternalId,
+  getMimeTypeFromGithubContentNodeType,
+  getRepositoryInternalId,
+} from "@connectors/connectors/github/lib/utils";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { concurrentExecutor } from "@connectors/lib/async_utils";
+import { upsertDataSourceFolder } from "@connectors/lib/data_sources";
+import {
+  GithubCodeDirectory,
+  GithubCodeFile,
+  GithubCodeRepository,
+  GithubDiscussion,
+  GithubIssue,
+} from "@connectors/lib/models/github";
+import { ConnectorModel } from "@connectors/resources/storage/models/connector_model";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+async function createFolderNodes() {
+  const connectors = await ConnectorModel.findAll({
```

Comment:
nit: I think we have a `listByType` on `ConnectorResource` that does exactly that, in general I use the Resource when there is one (doesn't matter here ofc)